app/src/main/java/com/github/sdp/ratemyepfl/model/items/Event.kt

```diff
@@ -19,53 +18,54 @@ class Event(
 ) : Reviewable(), Displayable {
 
     override fun toString(): String {
-        return id
+        return name
     }
 
+    override fun toHashMap(): HashMap<String, Any?> = hashMapOf<String, Any?>(
+        EventRepositoryImpl.NAME_FIELD_NAME to name,
+        EventRepositoryImpl.NUMBER_PARTICIPANTS_FIELD_NAME to numParticipants,
+        EventRepositoryImpl.LIMIT_PARTICIPANTS_FIELD_NAME to limitParticipants,
+        EventRepositoryImpl.LATITUDE_FIELD_NAME to lat,
+        EventRepositoryImpl.LONGITUDE_FIELD_NAME to long,
+        EventRepositoryImpl.DATE_FIELD_NAME to date.toString(),
+    ).apply { putAll(super.toHashMap()) }
+
     fun showParticipation(): String {
         return "Participants: $numParticipants/$limitParticipants"
     }
 
-    /**
-     * Creates an hash map of the Event, to add it to the DB
-     */
-    fun toHashMap(): HashMap<String, Any?> {
-        return hashMapOf(
-            EventRepository.NUMBER_PARTICIPANTS_FIELD_NAME to numParticipants.toString(),
-            EventRepository.LIMIT_PARTICIPANTS_FIELD_NAME to limitParticipants.toString(),
-            EventRepository.LATITUDE_FIELD_NAME to lat.toString(),
-            EventRepository.LONGITUDE_FIELD_NAME to long.toString(),
-            EventRepository.DATE_FIELD_NAME to date.toString(),
-            Repository.NUM_REVIEWS_FIELD_NAME to numReviews.toString(),
-            Repository.AVERAGE_GRADE_FIELD_NAME to averageGrade.toString()
-        )
-    }
+    override fun getId(): String = name
 
     override fun toMapItem(): MapItem {
         return EventItem(
             this,
-            MapActivityUtils.PHOTO_MAPPING.getOrDefault(id, R.raw.niki), // Arbitrary default value
+            MapActivityUtils.PHOTO_MAPPING.getOrDefault(

```

Comment:
Not your code, but we need to change this. Using Storage instead.

Reply:
How can I do this?

Reply:
It will take time, so it's a task for a next sprint.